lib/common/client/middleware/response/facility_parser.rb

```diff
@@ -33,30 +33,43 @@ def attribute_mappings(entry, mapping)
             attrs = entry['attributes']
             mapped_attrs = { 'address' => {}, 'services' => {} }
             mapping.slice('unique_id', 'name', 'classification', 'website').each do |key, value|
-              mapped_attrs[key] = attrs[value]
+              mapped_attrs[key] = clean(attrs[value])
             end
             %w[hours access feedback phone].each do |name|
               mapped_attrs[name] = nested(mapping[name], attrs)
             end
             mapped_attrs['address']['physical'] = nested(mapping['physical'], attrs)
             mapped_attrs['address']['mailing'] = nested(mapping['mailing'], attrs)
-            mapped_attrs['services']['benefits'] = nested(mapping['benefits'], attrs)
-            mapped_attrs['services']['last_updated'] = services_date(attrs)
-            mapped_attrs['services']['health'] = services_from_gis(mapping['services'], attrs)
-
-            mapped_attrs.merge!(
-              'lat' => entry['geometry']['y'],
-              'long' => entry['geometry']['x']
-            )
+            if mapping['benefits']
+              mapped_attrs['services']['benefits'] = {
+                'standard' => clean_benefits(nested(mapping['benefits'], attrs)),
+                'other' => attrs['Other_Services']
+              }
+
+            end
+            mapped_attrs['services']['last_updated'] = services_date(attrs) if mapping['services']
+            mapped_attrs['services']['health'] = services_from_gis(mapping['services'], attrs) if mapping['services']
+
+            mapped_attrs.merge!('lat' => entry['geometry']['y'], 'long' => entry['geometry']['x'])
           end
 
           def nested(item, attrs)
-            return unless item
+            return {} unless item
             item.each_with_object({}) do |(key, value), hash|
-              hash[key] = value.respond_to?(:call) ? value.call(attrs) : attrs[value]
+              hash[key] = value.respond_to?(:call) ? value.call(attrs) : clean(attrs[value])
```

Comment:
isn't there a method, `block_given?` commonly used with `yield` that would be more readable and idiomatic than `respond_to?(:call)` ?

Reply:
In this case, block_given returns false. It's probably possible to do something to my methods being called to change that, but couldn't find it right off. Given my vanilla methods, this might be the most readable way, but I'm always open to suggestions.